spec/controllers/attributes_controller_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require Rails.root.join('spec', 'support', 'mock_auth_helper.rb')
+
+describe AttributesController do
```

Comment:
Here are two places that give some explanation on why integration over controller:
http://betterspecs.org/#integration
http://stackoverflow.com/questions/6129694/should-i-really-test-controllers

I think that most of you controller tests could be converted to integration tests quickly. It seems like adding `type: :request` to the top `describe` and changing your request calls to something like `post 'attributes/create
, attribute: { key: 'foo', value: 'bar', resource_id: 1 } ` that should get you well on your way to make these tests include more of the stack. (I think that link on rails 5 integration tests was sent in haste, I didn't realize that was for minitest.)